### PR TITLE
Add chat ea1

### PR DESCRIPTION
**Pull request by Aichnerc**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [x] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently when not in a country chat, you can not join it.
- Currently after joining a country chat, the user cannot see the messages.
- Currently there is no loading animation for the chat page.
- Currently the holocron icon is thicc.
- Currently the register form is bugged where some of the error messages are bugged.
- Currently you can not see the preview of messages.
- Currently the chat card does not look different if there are unread messages.

**What is the new behavior (if this is a feature change)?**
- You can now join a country chat by clicking the `Join Chat` button.
- The message handler is now being created after joining the chat.
- A loading animation for the chat page has been added.
- The holocron icon has been resized to 50px, as required.
- The register page has been fixed.
- You can now see the preview of messages.
- The chat card now appears different if there are unread messages.

### CODE REVIEW

File: src/components/organisms/Chat/index.jsx [hunk 1]
```diff
@@ -62,13 +62,22 @@ class Chat extends React.Component {
...
+
   componentDidUpdate() {
     this.scrollToBottom();
   }
 
   componentWillUnmount = () => {
     // Stop getting messages for certain chat (close listener)
-    this.props.stopGettingMessages(this.props.chatDetails.id);
+    //this.props.stopGettingMessages(this.props.chatDetails.id);
   };
```
Comment (Theralun) [inline src/components/organisms/Chat/index.jsx:81]: should we get rid of this completely?

File: src/components/organisms/RegisterForm/index.jsx [hunk 2]
```diff
@@ -461,6 +462,8 @@ class HomePage extends React.Component {
       this.firstRow.current &&
       this.firstRow.current.scrollIntoView();
 
+    console.log(authError, authErrorCode);
+
```
Comment (schettn) [inline src/components/organisms/RegisterForm/index.jsx:466]: Is this needed?

Comment (Theralun) [inline src/components/organisms/RegisterForm/index.jsx:466]: useless

File: src/components/organisms/RegisterForm/index.jsx [hunk 3]
```diff
@@ -535,14 +538,15 @@ class HomePage extends React.Component {
...
+                    (authErrorCode === 1 || authErrorCode === 10) && (
+                      <small
+                        id="emailHelpError"
+                        className="form-text text-danger font-weight-bold"
+                      >
+                        {authError && authError}
+                      </small>
+                    )}
                   <small id="emailHelp" className="form-text text-muted">
                     Like your real name, your private E-Mail will not be shared.
```
Comment (schettn) [inline src/components/organisms/RegisterForm/index.jsx:551]: Is `E-Mail` correct? Shouldn't it be `e-mail`? Looks a bit germanish

File: src/components/pages/ChatPage/index.jsx [hunk 4]
```diff
@@ -257,6 +262,19 @@ class ChatPage extends React.Component {
     }
   };
 
+  getUserByUid = (uid) => {
+    const users = this.state.users;
+    let result = null;
```
Comment (schettn) [inline src/components/pages/ChatPage/index.jsx:267]: In snek we separate const and lets.
```suggestion
    const users = this.state.users;
    
    let result = null;
```

File: src/store/actions/authActions.js [hunk 5]
```diff
@@ -108,11 +108,29 @@ export const signUp = (newUser) => {
           return response.user.uid;
         })
         .catch((err) => {
-          dispatch({
-            type: "SIGNUP_ERROR",
-            errCode: 1,
-            err,
-          });
+          console.log(err.code);
```
Comment (schettn) [inline src/store/actions/authActions.js:111]: Still needed?

File: src/components/pages/ChatPage/index.jsx (hunk 4 above)
Comment (Aichnerc) [reply src/components/pages/ChatPage/index.jsx:267]: Makes sense, yeah.

**Pull request merged**